src/edit.js

```diff
@@ -36,47 +53,67 @@ export default function Edit( { attributes, setAttributes } ) {
 	useEffect( () => {
 		async function getRelatedPosts() {
 			const postId = window.location.href.match( /post=([\d]+)/ )[ 1 ];
```

Comment:
The `postId` can be received from the [context](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-context/) or, alternatively, using `useSelect()` hook (`select( 'core/editor' ).getCurrentPostId()`).

When using context (the preferred approach), you'd add the following entry to the `block.json`:

```json
"usesContext": [ "postId" ]
```

And then the `Edit` component would get a `context` prop (just like the `attributes` and `setAttributes`). You'd the destructure it in order to get the post ID:

```js
export default function Edit( { attributes, setAttributes, context } ) {
	const { countryCode, relatedPosts } = attributes;
	const { postId } = context;
```